ios/Classes/AblyFlutterPlugin.m

```diff
@@ -105,35 +96,43 @@ -(nullable AblyFlutter *)ablyWithHandle:(NSNumber *)handle;
 @implementation AblyFlutterPlugin {
     long long _nextRegistration;
     NSDictionary<NSString *, FlutterHandler>* _handlers;
-    NSNumber* _ablyHandle;
-    AblyFlutter* _ably;
+    AblyStreamsChannel* _streamsChannel;
     FlutterMethodChannel* _channel;
 }
 
+@synthesize ably = _ably;
+
 +(void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
     LOG(@"registrar: %@", [registrar class]);
+
+    // Initializing reader writer and method codecs
     FlutterStandardReaderWriter *const readerWriter = [AblyFlutterReaderWriter new];
     FlutterStandardMethodCodec *const methodCodec = [FlutterStandardMethodCodec codecWithReaderWriter:readerWriter];
     
-    FlutterMethodChannel *const channel =
-        [FlutterMethodChannel methodChannelWithName:@"io.ably.flutter.plugin" binaryMessenger:[registrar messenger] codec:methodCodec];
-    AblyFlutterPlugin *const plugin =
-        [[AblyFlutterPlugin alloc] initWithChannel:channel];
+    // initializing event channel for event listeners
+    AblyStreamsChannel *const streamsChannel = [AblyStreamsChannel streamsChannelWithName:@"io.ably.flutter.stream" binaryMessenger:registrar.messenger codec: methodCodec];
```

Comment:
For some of these longer lines could you consider formatting them so they're a little less long. I'm definitely not a fan of strict 80/120/whatever line length limits but there's always a sensible level - whether viewed in the IDE or a GitHub PR - when it's worth breaking them down. For Objective-C I think this is particularly easy to do, often aligned around argument colons. e.g.:

```suggestion
    AblyStreamsChannel *const streamsChannel =
        [AblyStreamsChannel streamsChannelWithName:@"io.ably.flutter.stream"
                                   binaryMessenger:registrar.messenger
                                             codec: methodCodec];
```

Reply:
Addressed in 86deac257bdc6054137ddedbfe31e85cd716412c